spec/intro.dd

```diff
@@ -8,59 +8,51 @@ $(SPEC_S Introduction,
         native executable.
         )
 
-$(H2 Phases of Compilation)
+$(H2 Conceptual models)
 
-$(P The process of compiling is divided into multiple phases. Each phase has no
-dependence on subsequent phases. For example, the scanner is not perturbed by
-the semantic analyzer. This separation of the passes makes language tools like
-syntax directed editors relatively easy to produce. It also is possible to
-compress D source by storing it in $(SINGLEQUOTE tokenized) form.)
+$(H3 Environments)
 
-$(OL
-        $(LI $(B source character set)$(BR)
+$(P The data-processing-system in which a D program is translated from source to executable format
+will be called the translation environment and the data-processing-system in which the translated
+D program is executed will be called the execution environment. The translation environment and
+the execution environment need not have the same system architecture.)
 
-        The source file is checked to see what character set it is,
-        and the appropriate scanner is loaded. ASCII and UTF
-        formats are accepted.
-        )
+$(H3 Program Structure)
 
-        $(LI $(B script line) $(BR)
+$(P A D program is kept in units called source files. Each source file contains a unit of translation
+called the module. D modules need not all be translated at the same time. Previously translated
+D modules may be individually preserved or in libraries. The separate modules of a program communicate
+by calls to functions whose identifers have external linkage, manipulation of objects whose identifers
+have external linkage, or manipulation of data files. Modules may be separately translated and then
+later linked to produce an executable program.)
 
-        If the first line starts with "$(HASH)!", then that line
-        is ignored.
-        )
+$(H3 Translation Phases)
```

Comment:
Compilation phases

Reply:
I renamed this to Translation Phases to keep it consistent with the terminology used in the C standard. The previous section I added uses the term 'translation' as it is used by the C standard.

Reply:
Thats probably because the C standard calls the unit of compilation a "translation unit". I don't know what we call them, if anything.

Reply:
I used this to describe it: `A D program is kept in units called source files. Each source file contains a unit of translation called the module. D modules need not all be translated at the same time.`

Reply:
Also code generation and optimization can span multiple modules so maybe compilation phases is better? 🤷‍♂

Reply:
Also, maybe this should be an H2? I'm not sure how this is a conceptual model

Reply:
Why would consistency with the C standard be a goal?

Reply:
It isn't. It just happens that in this instance it was useful to borrow some terminology.